packages/@o3r/workspace/src/cli/set-version.cts

```diff
@@ -46,20 +46,32 @@ program
 const options: any = program.opts();
 logger.level = options.verbose ? 'debug' : 'info';
 
-globbySync(options.include, {cwd: process.cwd()})
-  .map((file: string) => path.join(process.cwd(), file))
-  .map((filePath: string) => ({
-    path: filePath,
-    content: fs.readFileSync(filePath).toString()
-  }))
-  .forEach((pathWithContent: {path: string; content: string}) => {
-    const newContent = pathWithContent.content
-      .replace(new RegExp('"([~^]?)' + (options.placeholder as string).replace(/[.+?^${}()|[\]\\]/g, '\\$&').replace(/\\*\./g, '\\.') + '"', 'g'), `"$1${replaceVersion}"`)
-      .replace(/"workspace:([~^]?)[^"]*"(,?)$/gm, `"$1${replaceVersion}"$2`);
-    if (newContent !== pathWithContent.content) {
-      logger.info(`update version in ${pathWithContent.path}`);
-      fs.writeFileSync(pathWithContent.path, newContent);
-    } else {
-      logger.debug(`No change in ${pathWithContent.path}`);
-    }
-  });
+const cliFn = () => {
+  globbySync(options.include, {cwd: process.cwd()})
+    .map((file: string) => path.join(process.cwd(), file))
+    .map((filePath: string) => ({
+      path: filePath,
+      content: fs.readFileSync(filePath).toString()
+    }))
+    .forEach((pathWithContent: {path: string; content: string}) => {
+      const newContent = pathWithContent.content
+        .replace(new RegExp('"([~^]?)' + (options.placeholder as string).replace(/[.+?^${}()|[\]\\]/g, '\\$&').replace(/\\*\./g, '\\.') + '"', 'g'), `"$1${replaceVersion}"`)
+        .replace(/"workspace:([~^]?)[^"]*"(,?)$/gm, `"$1${replaceVersion}"$2`);
+      if (newContent !== pathWithContent.content) {
+        logger.info(`update version in ${pathWithContent.path}`);
+        fs.writeFileSync(pathWithContent.path, newContent);
+      } else {
+        logger.debug(`No change in ${pathWithContent.path}`);
+      }
+    });
+}
+
+void (async () => {
+  try {
+    const { createCliWithMetrics } = await import('@o3r/telemetry');
+    return createCliWithMetrics('@o3r/workspace:set-version', cliFn, { logger, preParsedOptions: options });
```

Comment:
example made once, I will wait for the approval to make the other ones in another PR